services/ui-src/src/utils/autosave/autosave.ts

```diff
@@ -33,49 +31,38 @@ export const autosaveFieldData = async ({
   fields,
   report,
   user,
-}: Props) => {
+}: AutosaveFieldDataProps) => {
   const { id, updateReport } = report;
   const { userName, state } = user;
 
-  // if field value hasn't changed from database value, don't autosave field
-  const onlyChangedFields = fields.filter(
-    // TODO: do we need a deeper equality check here?
-    (field: FieldInfo) => {
-      const { value, hydrationValue, defaultValue } = field;
-      return value !== defaultValue && !isEqual(value, hydrationValue);
-    }
+  // for each passed field, format for autosave payload (if changed)
+  const fieldsToSave: FieldDataTuple[] = await Promise.all(
+    fields
+      // filter to only fields with changed values
+      .filter((field: FieldInfo) => field.value !== field.hydrationValue)
+      // determine appropriate field value to set and return as tuple
+      .map(async (field: FieldInfo) => {
+        const { name, value, defaultValue, overrideCheck } = field;
+        const fieldValueIsValid = await form.trigger(name);
+        // if field value is valid or validity check overriden, use field value
+        if (fieldValueIsValid || overrideCheck) return [name, value];
+        // otherwise, revert field to default value
+        return [name, defaultValue];
+      })
```

Comment:
Note: Combined the filter and the map and simplified since we no longer need the additional checks.

Reply:
Thought: We might want to make this a separate method that we call from `autosaveFieldData` for both clarity and ease of testing.